cpp/ql/src/semmle/code/cpp/rangeanalysis/RangeSSA.qll

```diff
@@ -95,10 +95,19 @@ class RangeSsaDefinition extends ControlFlowNodeBase {
 
   /**
    * If this definition is a phi node corresponding to a guard,
-   * then return the variable and the guard.
+   * then return the variable access and the guard.
    */
-  predicate isGuardPhi(VariableAccess v, Expr guard, boolean branch) {
-    guard_defn(v, guard, this, branch)
+  predicate isGuardPhi(VariableAccess va, Expr guard, boolean branch) {
+    guard_defn(va, guard, this, branch)
+  }
+
+  /**
+   * If this definition is a phi node corresponding to a guard,
+   * then return the variable guarded, the variable access and the guard.
+   */
+  predicate isGuardPhi(StackVariable v, VariableAccess va, Expr guard, boolean branch) {
+    guard_defn(va, guard, this, branch) and
+    va.getTarget() = v
```

Comment:
I looked at the existing calls to `isGuardPhi`, and it turns out all 8 of them have a `getTarget() = v` right next to them. I suggest (in a separate commit or PR) that you deprecate the old 3-argument version of the predicate and update all callers.

Reply:
Addressed in https://github.com/github/codeql/pull/6678